go/arrow/util/protobuf_reflect_test.go

```diff
@@ -0,0 +1,156 @@
+	"google.golang.org/protobuf/types/known/anypb"
+	"reflect"
+	"testing"
+)
+
+func SetupTest() util_message.AllTheTypes {
+	msg := util_message.ExampleMessage{
+		Field1: "Example",
+	}
+
+	anyMsg, _ := anypb.New(&msg)
+
+	return util_message.AllTheTypes{
+		String_:     "Hello",
+		Int32:       10,
+		Int64:       100,
+		Sint32:      -10,
+		Sin64:       -100,
+		Uint32:      10,
+		Uint64:      100,
+		Fixed32:     10,
+		Fixed64:     1000,
+		Sfixed32:    10,
+		Bool:        false,
+		Bytes:       []byte("Hello, world!"),
+		Double:      1.1,
+		Enum:        0,
+		Message:     &msg,
+		Oneof:       &util_message.AllTheTypes_Oneofstring{Oneofstring: "World"},
+		Any:         anyMsg,
+		SimpleMap:   map[int32]string{99: "Hello", 100: "World"},
+		ComplexMap:  map[string]*util_message.ExampleMessage{"complex": &msg},
+		SimpleList:  []string{"Hello", "World"},
+		ComplexList: []*util_message.ExampleMessage{&msg},
+	}
+}
+
+func TestGetSchema(t *testing.T) {
+	msg := SetupTest()
+
+	pr := NewProtobufStructReflection(&msg)
+	got := pr.GetSchema().String()
+	want := `schema:
+  fields: 22
+    - string: type=utf8, nullable
+    - int32: type=int32, nullable
+    - int64: type=int64, nullable
+    - sint32: type=int32, nullable
+    - sin64: type=int64, nullable
+    - uint32: type=uint32, nullable
+    - uint64: type=uint64, nullable
+    - fixed32: type=uint32, nullable
+    - fixed64: type=uint64, nullable
+    - sfixed32: type=int32, nullable
+    - bool: type=bool, nullable
+    - bytes: type=binary, nullable
+    - double: type=float64, nullable
+    - enum: type=int32, nullable
+    - message: type=struct<field1: utf8>, nullable
+    - oneofstring: type=utf8, nullable
+    - oneofmessage: type=struct<field1: utf8>, nullable
+    - any: type=struct<field1: utf8>, nullable
+    - simple_map: type=map<int32, utf8, items_nullable>, nullable
+    - complex_map: type=map<utf8, struct<field1: utf8>, items_nullable>, nullable
+    - simple_list: type=list<item: utf8, nullable>, nullable
+    - complex_list: type=list<item: struct<field1: utf8>, nullable>, nullable`
+
+	require.Equal(t, want, got)
+
+	excludeComplex := func(pfr protobufFieldReflection) bool {
+		return pfr.isMap() || pfr.isList() || pfr.isStruct()
+	}
+
+	got = NewProtobufStructReflection(&msg, WithExclusionPolicy(excludeComplex)).GetSchema().String()
+	want = `schema:
+  fields: 15
+    - string: type=utf8, nullable
+    - int32: type=int32, nullable
+    - int64: type=int64, nullable
+    - sint32: type=int32, nullable
+    - sin64: type=int64, nullable
+    - uint32: type=uint32, nullable
+    - uint64: type=uint64, nullable
+    - fixed32: type=uint32, nullable
+    - fixed64: type=uint64, nullable
+    - sfixed32: type=int32, nullable
+    - bool: type=bool, nullable
+    - bytes: type=binary, nullable
+    - double: type=float64, nullable
+    - enum: type=int32, nullable
+    - oneofstring: type=utf8, nullable`
+
+	require.Equal(t, want, got)
```

Comment:
Looks like this test is failing in only *some* cases. You should try using `assert.Equalf(t, want, got, "wanted: %s\ngot: %s", want, got)` so that you can see the output and figure out why it's failing in some cases.